tests/contrib/test_gce.py

```diff
@@ -14,102 +14,195 @@
+            credentials._service_account_info,
+            {'email': A_SERVICE_ACCOUNT['email']}
+        )
+        self.assertEquals(credentials.scopes, A_SERVICE_ACCOUNT['scopes'])
+        self.assertEquals(
+            credentials.service_account_email,
+            A_SERVICE_ACCOUNT['email']
+        )
+        self.assertEquals(credentials.service_account_info, A_SERVICE_ACCOUNT)
+        get_metadata.assert_called_once_with(
+            path=[
+                'instance',
+                'service-accounts',
+                A_SERVICE_ACCOUNT['email']
+            ],
+            http_request=None
+        )
+
+    @mock.patch(METADATA_SERVER, return_value='a-project-id')
+    def test_project_id(self, get_metadata):
+        credentials = AppAssertionCredentials()
+        self.assertIsNone(credentials._project_id)
+        self.assertEquals(credentials.project_id, 'a-project-id')
+        self.assertEquals(credentials.project_id, 'a-project-id')
+        get_metadata.assert_called_once_with(
+            path=[
+                'project',
+                'project-id'
+            ],
+            recursive=False
+        )
+
+    def test_refresh_token(self):
         credentials = AppAssertionCredentials()
         self.assertEquals(None, credentials.access_token)
-        credentials.refresh(http)
-        self.assertEquals(access_token, credentials.access_token)
 
-        base_metadata_uri = (
-            'http://metadata.google.internal/computeMetadata/v1/instance/'
-            'service-accounts/default/token')
-        http.request.assert_called_once_with(
-            base_metadata_uri, headers={'Metadata-Flavor': 'Google'})
+        with mock.patch(METADATA_SERVER, return_value=ONE_SECOND_TOKEN):
+            credentials.get_access_token()
+            time.sleep(2)
+            self.assertTrue(credentials.access_token_expired)
+
+        with mock.patch(METADATA_SERVER, return_value=FOREVER_TOKEN):
+            credentials.get_access_token()
+            self.assertFalse(credentials.access_token_expired)
 
-    def test_refresh_success(self):
-        self._refresh_success_helper(bytes_response=False)
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_serialize_deserialize(self, get_metadata):
+        credentials = AppAssertionCredentials()
+        credentials_from_json = Credentials.new_from_json(
+            credentials.to_json()
+        )
+        self.assertTrue(
+            credentials.service_account_info,
+            credentials_from_json.service_account_info
+        )
+
+    @mock.patch('warnings.warn')
+    @mock.patch(METADATA_SERVER, return_value=A_SERVICE_ACCOUNT)
+    def test_create_scoped_valid(self, get_metadata, warn_mock):
+        scope = 'https://www.googleapis.com/auth/cloud-platform'
+        credentials = AppAssertionCredentials()
+        credentials.create_scoped(scope)
+        warn_mock.assert_called_once_with(_SCOPES_WARNING)
+
+    # ERROR TESTS
```

Comment:
Axe this comment. If you think the tests should be split up, just make a new `TestCase`.